zulip/zulip/__init__.py

```diff
@@ -512,6 +512,8 @@ def end_error_retry(succeeded):
                 if longpolling:
                     # When longpolling, we expect the timeout to fire,
                     # and the correct response is to just retry
+                    # after a safety waiting time.
+                    time.sleep(1)
```

Comment:
We don't need to wait here, because fundamentally, we just had a timeout, and timeouts are slow and unlikely to be a DoS risk.